### PR TITLE
porting tests from hand-rolled artisinal test-suite to mocha

### PR DESCRIPTION
**Pull request by bcoe**
__Note: this is a squashed version of https://github.com/NodeRedis/node_redis/tree/mocha-test-suite-no-grunt, since we're getting close to completing this task.__

We're on the home stretch of the work being done for #778, I would love to get some eyeballs on this.

reviewers: @erinspice, @blainsmith

### CODE REVIEW

**Comment by erinspice** [comment]
I love this so much. I will review it tomorrow.

**Comment by blainsmith** [comment]
So awesome. I'll get my eyeballs on it later today.

File: test/node_redis.spec.js [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+                            // For orderly shutdown in normal programs, do client.quit()
+                            client.stream.destroy();
+                        });
+                    });
+
+                    describe("and it's subscribed to a channel", function () {
+                        // reconnect_select_db_after_pubsub
+                        // Does not pass.
+                        // "Connection in subscriber mode, only subscriber commands may be used"
+                        xit("reconnects, unsubscribes, and can retrieve the pre-existing data", function (done) {
```
Comment (erinspice) [inline test/node_redis.spec.js]: This test is one I ported from the old suite. IIRC, it was formatted incorrectly in the old suite and was reporting passing when it should have been reporting failing. Does the description of this test describe behavior that should work, such that we need to enter an issue to fix the bug in node_redis, or is node_redis behaving as it should, such that we should delete this test?

**Comment by blainsmith** [comment]
Everything looks good to me. This will be a great addition.

**Comment by erinspice** [comment]
LGTM, too. Ship it!

File: test/node_redis.spec.js (hunk 1 above)
Comment (bcoe) [inline test/node_redis.spec.js]: Going to try to get this one test working, just finishing porting the final methods first.

**Pull request merged**